File: ksqldb/src/nodejs/consumer.js
const {Kafka} = require('kafkajs');

const kafka = new Kafka({
    clientId: 'demo-consumer',
    brokers: ['localhost:9092']
});

// Attention le group id doit être unique pour chaque consommateur
const consumer = kafka.consumer({groupId: 'demo-js-group'});


async function eachMessage({topic, partition, message}) {
    console.log({
        value: message.value.toString(),
        headers: message.headers,
    });
}

const run = async () => {
    await consumer.connect();
    await consumer.subscribe({topic: 'test'});
    await consumer.run({eachMessage});
}

run().catch(console.error);

File: ksqldb/src/python/consumer.py
# Notes sur la configuration du consumer:
# - Le consumer a besoin d'un group_id pour pouvoir reprendre la lecture du topic là où il s'était arrêté
from kafka import KafkaConsumer

consumer = KafkaConsumer(
    'output', 
    bootstrap_servers='localhost:9092',
    client_id='calc-consumer',
    group_id='calc-group',
    )

for event in consumer:
    if not event.value: 
        continue
    msg = event.value.decode('utf-8')
    key = event.key
    offset = event.offset
    print(f'Key: {key} | Message: {msg} | Offset: {offset}')